src/storage/index/bwtree_index.cpp

```diff
@@ -17,6 +17,34 @@ void BwTreeIndex<KeyType>::PerformGarbageCollection() {
   bwtree_->PerformGarbageCollection();
 }
 
+template <typename KeyType>
+size_t BwTreeIndex<KeyType>::EstimateHeapUsage() const {
+  // This is a back-of-the-envelope calculation that could be innacurate: it does not account for deltas within the
+  // BwTree. Also the mapping table is anonymously mmap'd so getting its exact size is tricky.
+  constexpr auto max_elements_per_node = 128;  // constant from bwtree.h
```

Comment:
Just curious, why not directly read the constant from `bwtree.h` but hard-code it this way?

Reply:
I could do that, but the question then becomes which value to read. There are two (inner node max size and leaf node max size) that currently happen to have the same value. I guess I could just read the leaf node size.